specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/IntegrationRuntime.json

```diff
@@ -1011,6 +1011,223 @@
+        }
+      ],
+      "properties": {
+        "folderId": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Folder id which contains environment."
+        },
+        "variables": {
+          "type": "array",
+          "description": "Variable in environment",
+          "items": {
+            "$ref": "#/definitions/SsisVariable"
+          }
+        }
+      }
+    },
+    "SsisParameter": {
+      "type": "object",
+      "description": "Ssis parameter.",
+      "properties": {
+        "id": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Parameter id."
+        },
+        "name": {
+          "type": "string",
+          "description": "Parameter name."
+        },
+        "description": {
+          "type": "string",
+          "description": "Parameter description."
+        },
+        "dataType": {
+          "type": "string",
+          "description": "Parameter type."
+        },
+        "required": {
+          "type": "boolean",
+          "description": "Whether parameter is reuqired."
+        },
+        "sensitive": {
+          "type": "boolean",
+          "description": "Whether parameter is sensitive."
+        },
+        "designDefaultValue": {
+          "type": "string",
+          "description": "Design default value of parameter."
+        },
+        "defaultValue": {
+          "type": "string",
+          "description": "Default value of parameter."
+        },
+        "sensitiveDefaultValue": {
+          "type": "string",
+          "description": "Default sensitive value of parameter."
+        },
+        "valueType": {
+          "type": "string",
+          "description": "Parameter value type."
+        },
+        "valueSet": {
+          "type": "boolean",
+          "description": "Parameter value set."
+        },
+        "variable": {
+          "type": "string",
+          "description": "Parameter reference variable."
+        }
+      }
+    },
+    "SsisVariable": {
+      "type": "object",
+      "description": "Ssis variable.",
+      "properties": {
+        "id": {
+          "type": "integer",
+          "format": "int64",
+          "description": "Variable id."
+        },
+        "name": {
+          "type": "string",
+          "description": "Variable name."
+        },
+        "description": {
+          "type": "string",
+          "description": "Variable description."
+        },
+        "dataType": {
+          "type": "string",
+          "description": "Variable type."
+        },
+        "sensitive": {
+          "type": "boolean",
+          "description": "Whether varaible is sensitive."
+        },
+        "value": {
+          "type": "string",
+          "description": "Variable value."
+        },
+        "sensitiveValue": {
+          "type": "string",
+          "description": "Variable sensitive value."
+        }
+      }
+    },
+    "SsisEnvironmentReference": {
```

Comment:
I can see you're adding new object definitions, but they don't appear to be referenced by any of the existing properties in this swagger. How are these new objects used?

Reply:
`SsisFolder`, `SsisProject`, `SsisPackage` and `SsisEnvironment` are subclass of `SsisObjectMetadata`. [IntegrationRuntimeObjectMetadata_Get](https://github.com/Azure/azure-rest-api-specs/blob/16a2ac3dfd26f79a49488411fa80734419763984/specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json#L1420) returns `SsisObjectMetadata` list, the elements belong to one of the subclass.
By adding these  subclass, customer can get the particular properties by casting. The elements in list must belong to same subclass.